nilearn/decoding/fista.py

```diff
@@ -184,7 +187,9 @@ def mfista(f1_grad, f2_prox, total_energy, lipschitz_constant, w_size,
 
         # backward (prox) step
         for _ in range(10):
-            w, prox_info = f2_prox(z - stepsize * gradient_buffer, stepsize,
+            # z <- -stepsize * gradient_buffer + z
+            z = inplace_add_mult(gradient_buffer, z, a=-stepsize)
```

Comment:
I recall from the ADMM version that this assignment to `z` was necessary there, too. If this operation really is in place, it wouldn't need the assignment to `z`. Does it work without it?